### PR TITLE
perf: prepeocess.py와 best_model 추가, train_model 변경

### PR DESCRIPTION
**Pull request by limegom**
200x200 pixel base 64 encoded 이미지 model input으로 변환 모델 테스트 결과 저장
모델 테스트 결과 저장

### CODE REVIEW

File: tests/best_model.pth
Comment (ljy2855) [inline tests/best_model.pth:1]: 테스트 코드에는 unittest 패키지를 통해서,
preprocess_base64_image 함수를 불러와 가상의 200x200 base64 encoding된 파일을 transform, model input 넣는 코드를 작성해주세요

**Review by ljy2855 — CHANGES_REQUESTED** [review]
👍

**Comment by limegom** [comment]
PS C:\usr\CSE4022\ML\tests> uv run python -m unittest discover                                                                
.
----------------------------------------------------------------------
Ran 1 test in 0.034s

OK
PS C:\usr\CSE4022\ML\tests> uv run python -m unittest discover
.
----------------------------------------------------------------------
Ran 1 test in 0.007s

OK

테스트 결과는 위에처럼 나와서 성공했고

train_model.py에
seed 추가
precision_score, recall_score, f1_score 추가
했습니다.

File: src/data/MNIST/raw/t10k-images-idx3-ubyte
Comment (ljy2855) [inline src/data/MNIST/raw/t10k-images-idx3-ubyte:1]: 어라 이거 gitignore로 처리했을텐데, 왜 올라왔을까요..?

File: src/train_model.py [hunk 1]
```diff
@@ -16,27 +26,29 @@ def get_device():
...
         print(f"Using device: {device}")
 
         train_loader, test_loader = get_dataloaders(batch_size=batch_size)
-        model = HybridCNN().to(device)
+        model = HybridCNN().to(device)   
         criterion = nn.CrossEntropyLoss()
         optimizer = optim.Adam(model.parameters(), lr=learning_rate, weight_decay=1e-4)
         scheduler = torch.optim.lr_scheduler.StepLR(optimizer, step_size=5, gamma=0.5)
-
+
```
Comment (ljy2855) [inline src/train_model.py:44]: 이런거들 자동으로 포맷팅 될텐데, 혹시 pre-commit 적용되어 있을까요?

**Review by ljy2855 — COMMENTED** [review]
data만 ignore로 추가하시고 merge 해주세요~

File: src/data/MNIST/raw/t10k-images-idx3-ubyte
Comment (limegom) [reply src/data/MNIST/raw/t10k-images-idx3-ubyte:1]: 제가 sourcetree로 그냥 올려서 혹시 이것도 올려야되나? 싶어서
그냥 스테이지에 올리니까 push되서 같이 올렸습니다.
삭제하고 다시 push할까요?

Comment (ljy2855) [reply src/data/MNIST/raw/t10k-images-idx3-ubyte:1]: 넴 부탁드림니다 👍

File: src/train_model.py (hunk 1 above)
Comment (ljy2855) [reply src/train_model.py:44]: `pre-commit install` 한번 확인부탁드랴요~

Comment (limegom) [reply src/train_model.py:44]: pip install pre-commit
하니까 받아졌습니다!
